src/Sticky.jsx

```diff
@@ -170,17 +170,23 @@ class Sticky extends Component {
     }
 
     handleResize (e, ae) {
+        if (this.props.shouldFreeze()) { return; }
+
         winHeight = ae.resize.height;
         this.updateInitialDimension();
         this.update();
     }
 
     handleScrollStart (e, ae) {
+        if (this.props.shouldFreeze()) { return; }
+        
         scrollTop = ae.scroll.top;
         this.updateInitialDimension();
     }
 
     handleScroll (e, ae) {
+        if (this.props.shouldFreeze()) { return; }
```

Comment:
Actually I am going with a simple object property rather than state, since we don't want to trigger re-render like you said.